scylla-cdc/src/reader.rs

```diff
@@ -0,0 +1,266 @@
+        let sleep_interval: time::Duration = time::Duration::new(SLEEP_INTERVAL, 0);
+        let window_size: chrono::Duration = chrono::Duration::seconds(WINDOW_SIZE);
+        let safety_interval: chrono::Duration = chrono::Duration::seconds(SAFETY_INTERVAL);
+
+        let reader = Reader::test_new(
+            Vec::from([stream_id]),
+            start_timestamp,
+            window_size,
+            safety_interval,
+            sleep_interval,
+        );
+
+        Ok(reader)
+    }
+
+    fn get_create_table_query() -> String {
+        format!("CREATE TABLE {}.{} (pk int, t int, v text, s text, PRIMARY KEY (pk, t)) WITH cdc = {{'enabled':true}};",
+                TEST_KEYSPACE,
+                TEST_TABLE
+        )
+    }
+
+    async fn create_test_db(session: &Session) {
+        let mut create_keyspace_query = Query::new(format!(
+            "CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class': 'SimpleStrategy', 'replication_factor': 1}};",
+            TEST_KEYSPACE
+        ));
+        create_keyspace_query.set_consistency(Consistency::All);
+        session.query(create_keyspace_query, &[]).await.unwrap();
+
+        // Create test table
+        let create_table_query = get_create_table_query();
+        session.query(create_table_query, &[]).await.unwrap();
+    }
+
+    async fn populate_test_db(session: &Session) {
+        session
+            .query(format!("use {};", TEST_KEYSPACE), ())
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 0, 'val0', 'static0');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 1, 'val1', 'static1');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+        session
+            .query(
+                format!(
+                    "INSERT INTO {} (pk, t, v, s) VALUES (0, 2, 'val2', 'static2');",
+                    TEST_TABLE
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+    }
+
+    async fn get_cdc_stream_id(session: &Session) -> Result<StreamID, Box<dyn Error>> {
+        let query_stream_id = format!(
+            "SELECT \"cdc$stream_id\" FROM {}_scylla_cdc_log limit 1;",
+            TEST_TABLE
+        );
+
+        let mut stream_id_vec: StreamID = vec![];
+        let mut rows = session
+            .query_iter(query_stream_id, &[])
+            .await?
+            .into_typed::<(StreamID,)>();
+
+        while let Some(next_row) = rows.next().await {
+            stream_id_vec = next_row.unwrap().0;
+        }
+        Ok(stream_id_vec)
```

Comment:
Are you sure you want to loop here?
`Ok(rows.next().await.unwrap()?.0)` should do the trick as well